grid2op/Reward/_alertTrustScore.py

```diff
@@ -0,0 +1,168 @@
+    where the agent is asked to send information about potential line overload issue on the grid after unpredictable powerline
+    disconnection (attack of the opponent).
+    The alerts are assessed once per attack and cumulated over the episode. In this scheme, this "reward" computed the assistant "score",
+    which assesses how well the agent is aware of its capacities to deal with a situation during an episode.
+    It should not be used to train an agent.
+    
+    """
+
+    #TODO
+    #Parameters to use for challenge
+    #def __init__(self,
+    #             logger=None,
+    #             reward_min_no_blackout=-1.0,
+    #             reward_min_blackout=-50,
+    #             reward_max_no_blackout=0.0,
+    #             reward_max_blackout=0.0,
+    #             reward_end_episode_bonus=0.0,
+    #             min_score=-3):
+    def __init__(self,
+                 logger=None,
+                 reward_min_no_blackout=-1.0,
+                 reward_min_blackout=-10.0,
+                 reward_max_no_blackout=1.0,
+                 reward_max_blackout=2.0,
+                 reward_end_episode_bonus=1.0,
+                 min_score=-1.0):
+        
+        super().__init__(logger,
+                 reward_min_no_blackout,
+                 reward_min_blackout, 
+                 reward_max_no_blackout,
+                 reward_max_blackout,
+                 reward_end_episode_bonus)
+        
+        self.min_score = dt_float(min_score)
+        self.max_score = dt_float(1.0)
+        
+    def initialize(self, env):
+        self._is_simul_env = self.is_simulated_env(env)
+        #SURVIVOR_TIMESTEPS=env._al
+            
+        self.cumulated_reward = 0
+        #KPIs
+        self.total_nb_attacks = 0
+        self.nb_last_attacks = 0 #attacks in the AlertTimeWindow before done
+
+        #TODO
+        #self.total_nb_alerts = 0
+        #self.alert_attack_no_blackout = 0
+        #self.alert_attack_blackout = 0
+        #self.no_alert_attack_no_blackout = 0
+        #self.no_alert_attack_blackout = 0
+        #self.blackout_encountered = False
+        return super().initialize(env)
+            
+    def reset(self, env):
+        super().reset(env)
+        self.cumulated_reward = 0
+        #KPIs
+        self.total_nb_attacks = 0
+        self.nb_last_attacks=0
+
+        # TODO
+        #self.total_nb_alerts = 0
+        #self.alert_attack_no_blackout = 0
+        #self.alert_attack_blackout = 0
+        #self.no_alert_attack_no_blackout = 0
+        #self.no_alert_attack_blackout = 0
+        #self.blackout_encountered = False
+            
+    def __call__(self, action, env, has_error, is_done, is_illegal, is_ambiguous):
+
+        self.blackout_encountered = self.is_in_blackout(has_error, is_done)
+        
+        score_ep = 0.
+        if self._is_simul_env:
+            return score_ep
```

Comment:
Do this before anything else (faster)

Reply:
You mean in the __call__ method ?

Like this before the blackout_encountered line ?
![image](https://github.com/rte-france/Grid2Op/assets/5496312/cb9076df-c2d5-47eb-9da0-fbc88ac5f439)

Reply:
yes I think so